2697. Lexicographically Smallest Palindrome/solution.py

```diff
@@ -0,0 +1,20 @@
+class Solution:
+    # takes in the string s and turns it into a palindrome
+    # by replacing as few letters as possible. breaks tie by
+    # checking the lexicographical value of the current letter.
+    # returns the palindrome string.
+    def makeSmallestPalindrome(self, s: str) -> str:
+        # turn the string into a list for easier manipulation
+        
+        # iterate through the string starting from 0 and ending at half the length (rounded down)
+        # the reason why we're iterating to half the length is because we already
+        # know what the other half of the characters we are iterating over are
```

Comment:
Do you mean we can use the negative indices in Perl to address the other half? You can make that more clear in your comment.

Reply:
Yes, technically. ...But what is Perl?

Reply:
Sorry I meant Python, not Perl lol.